charts/virtual-nodes/templates/nodes.yaml

```
@@ -0,0 +1,132 @@
+{{- $params := dict }}
+
+# The default set of resources for all node types
+{{- $defaultResources := dict }}
+{{- $defaultResources = set $defaultResources "hugepages-1Gi" 0   }}
+{{- $defaultResources = set $defaultResources "hugepages-2Mi" 0   }}
+{{- $defaultResources = set $defaultResources "pods"          110 }}
+
+# dgxa100.40g
+# Specs taken from here:
+# https://docs.nvidia.com/dgx/dgxa100-user-guide/introduction-to-dgxa100.html#hardware-overview
+{{- if eq $node.type "dgxa100.40g" }}
+  {{- $resources := deepCopy $defaultResources }}
+  {{- $resources = set $resources "cpu"                 256    }}
+  {{- $resources = set $resources "memory"              "1Ti"  }}
+  {{- $resources = set $resources "nvidia.com/gpu"      8      }}
+  {{- $resources = set $resources "nvidia.com/mlnxnics" 16     }}
+  {{- $resources = set $resources "ephemeral-storage"   "15Ti" }}
+  {{- $params = set $params "resources" $resources }}
+
+# dgxa100.80g
+# Specs taken from here:
+# https://docs.nvidia.com/dgx/dgxa100-user-guide/introduction-to-dgxa100.html#hardware-overview
+{{- else if eq $node.type "dgxa100.80g" }}
+  {{- $resources := deepCopy $defaultResources }}
+  {{- $resources = set $resources "cpu"                 256    }}
+  {{- $resources = set $resources "memory"              "2Ti"  }}
+  {{- $resources = set $resources "nvidia.com/gpu"      8      }}
+  {{- $resources = set $resources "nvidia.com/mlnxnics" 16     }}
+  {{- $resources = set $resources "ephemeral-storage"   "30Ti" }}
+  {{- $params = set $params "resources" $resources }}
+
+# dgxh100.80g
+# Specs taken from here:
+# https://docs.nvidia.com/dgx/dgxh100-user-guide/introduction-to-dgxh100.html#hardware-overview
+{{- else if eq $node.type "dgxh100.80g" }}
+  {{- $resources := deepCopy $defaultResources }}
+  {{- $resources = set $resources "cpu"                 224    }}
+  {{- $resources = set $resources "memory"              "2Ti"  }}
+  {{- $resources = set $resources "nvidia.com/gpu"      8      }}
+  {{- $resources = set $resources "nvidia.com/mlnxnics" 16     }}
+  {{- $resources = set $resources "ephemeral-storage"   "30Ti" }}
+  {{- $params = set $params "resources" $resources }}
+
+# cpu.x86
+{{- else if eq $node.type "cpu.x86" }}
+  {{- $resources := deepCopy $defaultResources }}
+  {{- $resources = set $resources "cpu"               48             }}
+  {{- $resources = set $resources "memory"            "196692052Ki"  }}
+  {{- $resources = set $resources "ephemeral-storage" "2537570228Ki" }}
+  {{- $params = set $params "resources" $resources }}
+
+{{- else }}
+  {{- $error := printf "Unsupported node type '%s'" $node.type }}
+  {{- fail $error }}
+{{- end }}
+
+{{- $count := ($node.count | int) }}
+{{- range until $count }}
+---
+apiVersion: v1
+kind: Node
+metadata:
+  annotations:
+    node.alpha.kubernetes.io/ttl: "0"
+    kwok.x-k8s.io/node: fake
+    {{- range $key, $value := $node.annotations }}
+      {{ $key }}: {{ $value | quote }}
```

I think the spacing needs to be similar to labels 

```suggestion
    {{ $key }}: {{ $value | quote }}
```

---

addressed, thanks!